test/minikube/minikube_base_restore_test.go

```diff
@@ -436,3 +435,91 @@ func TestKubernetesAutoRestore(t *testing.T) {
 	})
 }
+
+// Basic test for creating a database off of a VolumeSnapshot
+func TestKubernetesSnapshotRestore(t *testing.T) {
+	testlib.AwaitTillerUp(t)
+	defer testlib.VerifyTeardown(t)
+
+	options := helm.Options{}
+
+	defer testlib.Teardown(testlib.TEARDOWN_ADMIN)
+
+	helmChartReleaseName, namespaceName := testlib.StartAdmin(t, &options, 1, "")
+
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
+	admin := fmt.Sprintf("%s-nuodb-cluster0", helmChartReleaseName)
+	admin0 := fmt.Sprintf("%s-0", admin)
+
+	testlib.AddDiagnosticTeardown(testlib.TEARDOWN_ADMIN, t, func() {
+		k8s.RunKubectl(t, kubectlOptions, "get", "pods", "-o", "wide")
+		testlib.DescribePods(t, namespaceName, admin)
+	})
+	defer testlib.Teardown(testlib.TEARDOWN_DATABASE)
+
+	sourceDatabaseChartName := testlib.StartDatabase(t, namespaceName, admin0, &helm.Options{
+		SetValues: map[string]string{
+			"database.sm.resources.requests.cpu":                       testlib.MINIMAL_VIABLE_ENGINE_CPU,
+			"database.sm.resources.requests.memory":                    testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"database.te.resources.requests.cpu":                       testlib.MINIMAL_VIABLE_ENGINE_CPU,
+			"database.te.resources.requests.memory":                    testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"database.persistence.storageClass":                        testlib.SNAPSHOTABLE_STORAGE_CLASS,
+			"database.sm.hotCopy.journalPath.persistence.storageClass": testlib.SNAPSHOTABLE_STORAGE_CLASS,
+			"database.sm.hotCopy.journalPath.enabled":                  "true",
+		},
+	})
+
+	output, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", admin0, "-c", "admin", "--", "bash", "-c",
+		"echo \"CREATE TABLE testtbl (id INT); INSERT INTO testtbl (id) values (123);\" | nuosql demo --user dba --password secret")
+
+	require.NoError(t, err, output)
+
+	smPod := fmt.Sprintf("sm-%s-nuodb-cluster0-demo-hotcopy-0", sourceDatabaseChartName)
+	output, err = k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", smPod, "-c", "engine", "--", "bash", "-c",
+		"echo \"{\\\"id\\\": \\\"123abc\\\"}\" > /var/opt/nuodb/archive/nuodb/demo/backup.json")
+	require.NoError(t, err, output)
+
+	output, err = k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", smPod, "-c", "engine", "--", "bash", "-c",
+		"echo \"{\\\"id\\\": \\\"123abc\\\"}\" > /var/opt/nuodb/journal/nuodb/demo/backup.json")
+	require.NoError(t, err, output)
+
+	defer testlib.Teardown(testlib.TEARDOWN_SNAPSHOT)
+
+	achiveVolumeName := "archive-volume-" + smPod
+	archiveSnapshotName := "archive-snapshot"
+	testlib.SnapshotVolume(t, namespaceName, achiveVolumeName, archiveSnapshotName)
+
+	journalVolumeName := "journal-volume-" + smPod
+	journalSnapshotName := "journal-snapshot"
+	testlib.SnapshotVolume(t, namespaceName, journalVolumeName, journalSnapshotName)
+
+	helm.DeleteE(t, &helm.Options{KubectlOptions: k8s.NewKubectlOptions("", "", namespaceName)}, sourceDatabaseChartName, true)
+
+	restoredDb := "db-clone"
```

Comment:
Is there a test where we exercise creation of a database with the same name as the snapshotted database?

Reply:
No. Do we have a good way to clear out left over PVCs?

Reply:
The PVC names should be predictable based on the archive names appearing in `volumeClaimTemplates`. You can just delete `"archive-volume-" + smPodName`.